source/dotnet/Library/AdaptiveCards/AdaptiveActionSet.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using System.Xml.Serialization;
+
+namespace AdaptiveCards
+{
+    /// <summary>
+    ///     ActionSet
+    /// </summary>
+#if !NETSTANDARD1_3
+    [XmlType(TypeName = AdaptiveActionSet.TypeName)]
+#endif
+    public class AdaptiveActionSet : AdaptiveElement
+    {
+        public const string TypeName = "ActionSet";
+
+#if !NETSTANDARD1_3
+        [XmlIgnore]
+#endif
+        [JsonProperty(Required = Required.Default)]
+        public override string Type { get; set; } = TypeName;
+
+        /// <summary>
+        ///     Actions of the action set
+        /// </summary>
+        [JsonConverter(typeof(IgnoreEmptyItemsConverter<AdaptiveAction>))]
+#if !NETSTANDARD1_3
+        [XmlArray("Actions")]
+        [XmlArrayItem(ElementName = "OpenUrl", Type = typeof(AdaptiveOpenUrlAction))]
+        [XmlArrayItem(ElementName = "ShowCard", Type = typeof(AdaptiveShowCardAction))]
+        [XmlArrayItem(ElementName = "Submit", Type = typeof(AdaptiveSubmitAction))]
+        [XmlArrayItem(ElementName = "ToggleVisibility", Type = typeof(AdaptiveToggleVisibilityAction))]
```

Comment:
I'm not entirely familiar with how this works, but does this explicit list interfere with supporting the custom action scenario in this case?

Reply:
I took this part of code from the AdaptiveCard class 

just talked with Carlos, apparently there's some support for custom actions added in the falllback support (will have to check if there's actual custom action support) so when fallback PR is merged, I'll update this section

---
In reply to: [272653316](https://github.com/Microsoft/AdaptiveCards/pull/2614#discussion_r272653316) [](ancestors = 272653316)

Reply:
Sounds good. If that ends up being a separate change, could you log an issue so we don't lose track of it?

---
In reply to: [273178556](https://github.com/Microsoft/AdaptiveCards/pull/2614#discussion_r273178556) [](ancestors = 273178556,272653316)

Reply:
https://github.com/Microsoft/AdaptiveCards/issues/2676